docs/concepts/workloads/controllers/deployment.md

```diff
@@ -227,6 +227,21 @@ killing the 3 `nginx:1.7.9` Pods that it had created, and will start creating
 `nginx:1.9.1` Pods. It will not wait for 5 replicas of `nginx:1.7.9` to be created
 before changing course.
 
+### Label selector updates
+
+It is generally discouraged to make label selector updates and it is suggested to plan your selectors up front.
+In any case, if you need to perform a label selector update, exercise great caution and make sure you have grasped
+all of the implications.
+
+* Selector additions require the pod template labels in the Deployment spec to be updated with the new label, too,
+otherwise a validation error is returned. This change is a non-overlapping one, meaning that the new selector does
+not select ReplicaSets and Pods created with the old selector, resulting in orphaning all of those objects. Also,
+note that changes in the PodTemplateSpec of the Deployment result in new rollouts, but a new ReplicaSet with Pods
```

Comment:
Yes, I meant that these were separate points:

1. Changes to PodTemplateSpec results in new rollouts
2. New ReplicaSet will be created since it isn't matching anything because of selector addition.

I think it just seemed a bit difficult to comprehend why those were stated together on my first pass.